doc/index.rst

```diff
@@ -0,0 +1,600 @@
+   improved in 3.11; ``typing_extensions`` backports these changes.
+
+   .. versionchanged:: 4.6.0
+
+      The improvements from Python 3.10 and 3.11 were backported.
+
+.. data:: NoReturn
+
+   See :py:data:`typing.NoReturn`. In ``typing`` since 3.5.4 and 3.6.2.
+
+.. data:: NotRequired
+
+   See :py:data:`typing.NotRequired`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.0.0
+
+.. class:: ParamSpec(name, *, default=...)
+
+   See :py:class:`typing.ParamSpec`. In ``typing`` since 3.10.
+
+   The ``typing_extensions`` version adds support for the
+   ``default=`` argument from :pep:`696`.
+
+   .. versionchanged:: 4.4.0
+
+      Added support for the ``default=`` argument.
+
+   .. versionchanged:: 4.6.0
+
+      The implementation was changed for compatibility with Python 3.12.
+
+.. class:: ParamSpecArgs
+
+.. class:: ParamSpecKwargs
+
+   See :py:class:`typing.ParamSpecArgs` and :py:class:`typing.ParamSpecKwargs`.
+   In ``typing`` since 3.10.
+
+.. class:: Protocol
+
+   See :py:class:`typing.Protocol`. In ``typing`` since 3.8.
+
+   Python 3.12 improves the performance of runtime-checkable protocols;
+   ``typing_extensions`` backports this improvement.
+
+   .. versionchanged:: 4.6.0
+
+      Backported the ability to define ``__init__`` methods on Protocol classes.
+
+   .. versionchanged:: 4.6.0
+
+      Backported changes to runtime-checkable protocols from Python 3.12,
+      including :py:gh:`103034` and :py:gh:`26067`.
+
+.. data:: Required
+
+   See :py:data:`typing.Required`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.0.0
+
+.. data:: Self
+
+   See :py:data:`typing.Self`. In ``typing`` since 3.11.
+
+   .. versionadded:: 4.0.0
+
+.. class:: Type
+
+   See :py:class:`typing.Type`. In ``typing`` since 3.5.2.
+
+.. data:: TypeAlias
+
+   See :py:data:`typing.TypeAlias`. In ``typing`` since 3.10.
+
+.. class:: TypeAliasType(name, value, *, type_params=())
+
+   See :py:class:`typing.TypeAliasType`. In ``typing`` since 3.12.
+
+   .. versionadded:: 4.6.0
+
+.. data:: TypeGuard
+
+   See :py:data:`typing.TypeGuard`. In ``typing`` since 3.10.
+
+.. class:: TypedDict
+
+   See :py:class:`typing.TypedDict`. In ``typing`` since 3.8.
+
+   ``typing_extensions`` backports various bug fixes and improvements
+   to ``TypedDict`` on Python 3.11 and lower.
+   :py:class:`TypedDict` does not store runtime information
+   about which (if any) keys are non-required in Python 3.8, and does not
+   honor the ``total`` keyword with old-style ``TypedDict()`` in Python
+   3.9.0 and 3.9.1. :py:class:`typing.TypedDict` also does not support multiple inheritance
+   with :py:class:`typing.Generic` on Python <3.11, and :py:class:`typing.TypedDict` classes do not
+   consistently have the ``__orig_bases__`` attribute on Python <3.12. The
+   ``typing_extensions`` backport provides all of these features and bugfixes on
+   all Python versions.
+
+   .. versionadded:: 4.3.0
```

Comment:
Don't know when it was added, but it wasn't in 4.3.0 :)

```suggestion
   .. versionchanged:: 4.3.0
```

Reply:
Thanks, just pushed that